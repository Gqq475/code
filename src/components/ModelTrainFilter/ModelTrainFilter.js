import React from 'react'
import classes from './ModelTrainFilter.scss'
import DatePicker2 from 'components/DatePicker2'
import AvailableFilters from 'components/AvailableFilters'
import { Button } from '@blueprintjs/core'
import { dateHOC } from '../../Utils'

const formatDate = (date) => date.format('YYYY-MM-DD HH:mm:ss')

type Props = {
  getModelTrainList: Function,
  dateStart: String,
  dateEnd: String
};
export class ModelTrainFilter extends React.Component {
  props: Props

  constructor (props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
    this.handleChange = this.handleChange.bind(this)
    this.handleChangeStartReliability = this.handleChangeStartReliability.bind(this)
    this.filterData = this.filterData.bind(this)
    this.state = {
      product: '',
      operation: '',
      lineID: '',
      subEntity: '',
      reliabilitys: [0, 100],
      pageNo: 1,
      // Must set 1 to get total count
      pageSize: 1,
      sortBy: 'id',
      order: 'asc'
    }
  }

  filterData () {
    const filterData = {}
    const stateKeys = Object.keys(this.state)
    const stateValues = Object.values(this.state)

    stateKeys.forEach((key, i) => {
      let value = stateValues[i]
      if (value && key === 'reliability') {
        value = (parseInt(value) / 100)
        value === 1 && (value = '')
      }
      value && (filterData[key] = value)
    })

    return filterData
  }

  handleClick () {
    this.props.getModelTrainList({
      dateStart: formatDate(this.props.dateStart),
      dateEnd: formatDate(this.props.dateEnd)
    })
  }

  componentDidMount () {
    this.props.getModelTrainList({
      dateStart: formatDate(this.props.dateStart),
      dateEnd: formatDate(this.props.dateEnd)
    })
  }

  handleChange (type) {
    return (e) => {
      this.setState({ [type]: e.target.value })
    }
  }

  handleChangeStartReliability () {
    return (reliabilitys) => {
      this.setState({reliabilitys})
    }
  }

  render () {
    return (
      <div className={classes['ModelTrainFilter-container']}>
        <DatePicker2 {...this.props} />
        <AvailableFilters {...this.state} that={this} />
        <Button className={classes['btn-submit']} onClick={this.handleClick}>
          查询
        </Button>
      </div>
    )
  }
}

export default dateHOC(props => <ModelTrainFilter {...props} />)
