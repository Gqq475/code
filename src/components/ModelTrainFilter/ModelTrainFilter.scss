.ModelTrainFilter-container {

}

.btn-submit {
  margin: 20px auto 0;
  display: block;
}
