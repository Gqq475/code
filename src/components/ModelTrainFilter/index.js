import ModelTrainFilter from './ModelTrainFilter'
export default ModelTrainFilter
