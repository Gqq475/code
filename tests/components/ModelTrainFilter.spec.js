import React from 'react'
import { ModelTrainFilter } from 'components/ModelTrainFilter/ModelTrainFilter'

describe('(Component) ModelTrainFilter', () => {
  it('render normally', () => {
    const wrapper = shallow(
      <ModelTrainFilter />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
