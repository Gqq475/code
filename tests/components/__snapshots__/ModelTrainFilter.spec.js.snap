// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ModelTrainFilter render normally 1`] = `
<div
  className="ModelTrainFilter-container"
>
  <DatePicker2 />
  <AvailableFilters
    lineID=""
    operation=""
    order="asc"
    pageNo={1}
    pageSize={1}
    product=""
    reliabilitys={
      Array [
        0,
        100,
      ]
    }
    sortBy="id"
    subEntity=""
    that={
      ModelTrainFilter {
        "_reactInternalInstance": ShallowComponentWrapper {
          "_calledComponentWillUnmount": false,
          "_compositeType": 0,
          "_context": Object {},
          "_currentElement": <ModelTrainFilter />,
          "_debugID": 1,
          "_hostContainerInfo": null,
          "_hostParent": null,
          "_instance": [Circular],
          "_mountOrder": 1,
          "_pendingCallbacks": null,
          "_pendingElement": null,
          "_pendingForceUpdate": false,
          "_pendingReplaceState": false,
          "_pendingStateQueue": null,
          "_renderedComponent": NoopInternalComponent {
            "_currentElement": <div
              className="ModelTrainFilter-container"
      >
              <DatePicker2 />
              <AvailableFilters
                      lineID=""
                      operation=""
                      order="asc"
                      pageNo={1}
                      pageSize={1}
                      product=""
                      reliabilitys={
                              Array [
                                      0,
                                      100,
                                    ]
                      }
                      sortBy="id"
                      subEntity=""
                      that={[Circular]}
              />
              <Blueprint.Button
                      className="btn-submit"
                      onClick={[Function]}
              >
                      查询
              </Blueprint.Button>
      </div>,
            "_debugID": 2,
            "_renderedOutput": <div
              className="ModelTrainFilter-container"
      >
              <DatePicker2 />
              <AvailableFilters
                      lineID=""
                      operation=""
                      order="asc"
                      pageNo={1}
                      pageSize={1}
                      product=""
                      reliabilitys={
                              Array [
                                      0,
                                      100,
                                    ]
                      }
                      sortBy="id"
                      subEntity=""
                      that={[Circular]}
              />
              <Blueprint.Button
                      className="btn-submit"
                      onClick={[Function]}
              >
                      查询
              </Blueprint.Button>
      </div>,
          },
          "_renderedNodeType": 0,
          "_rootNodeID": 0,
          "_topLevelWrapper": null,
          "_updateBatchNumber": null,
          "_warnedAboutRefsInRender": false,
        },
        "context": Object {},
        "filterData": [Function],
        "handleChange": [Function],
        "handleChangeStartReliability": [Function],
        "handleClick": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "lineID": "",
          "operation": "",
          "order": "asc",
          "pageNo": 1,
          "pageSize": 1,
          "product": "",
          "reliabilitys": Array [
            0,
            100,
          ],
          "sortBy": "id",
          "subEntity": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      }
    }
  />
  <Blueprint.Button
    className="btn-submit"
    onClick={[Function]}
  >
    查询
  </Blueprint.Button>
</div>
`;
